.github/workflows/publish-docs.yml

```diff
@@ -0,0 +1,34 @@
+name: Publish MkDocs on Main Branch
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: 3.8
+
+    - name: Install dependencies
+      run: |
+        python3 -m pip install --upgrade pip
+        pip install poetry
+        poetry install --with=dev
+
+    - name: Deploy to GitHub Pages
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        git config --global user.name "GitHub Actions Bot"
+        git config --global user.email "github-actions[bot]@users.noreply.github.com"
+        poetry run mike deploy --push --force --message "Deployed by GitHub Actions" main
```

Comment:
**Issue:** It looks like we're missing the `mkdocs-material` dependency for the documentation site, and the `mike` dependency for managing the deployment.

**Suggestion:** I'd recommend adding a new dependency group called `docs` that has all of the deps required to build the docs site, and then update the `Install dependencies` step on line 26 to install the new `docs` group instead of the `dev` group.